File: apps/api/app/schemas/project.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ProjectCreate(BaseModel):
    name: str
    description: str | None = None
    aspect_ratio: str
    width: int
    height: int
    fps: int


class ProjectUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
    aspect_ratio: str | None = None
    width: int | None = None
    height: int | None = None
    fps: int | None = None


class ProjectRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    name: str
    description: str | None
    aspect_ratio: str
    width: int
    height: int
    fps: int
    created_at: datetime
    updated_at: datetime

File: apps/web/src/types/project.ts
export type Project = {
  id: string
  name: string
  description: string | null
  aspect_ratio: string
  width: number
  height: number
  fps: number
  created_at: string
  updated_at: string
}

export type ProjectCreate = {
  name: string
  description: string
  aspect_ratio: string
  width: number
  height: number
  fps: number
}

export type ProjectExportFile = {
  scene_id: string
  scene_number: number
  asset_id: string
  filename: string
}

export type ProjectExportResult = {
  project_id: string
  export_id: string
  export_dir: string
  manifest_filename: string
  files: ProjectExportFile[]
}
